app/src/main/java/es/wolfi/app/passman/CustomFieldEditAdapter.java

```diff
@@ -0,0 +1,282 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package es.wolfi.app.passman;
+
+import android.annotation.SuppressLint;
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.graphics.Rect;
+import android.os.Build;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.DisplayMetrics;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import androidx.appcompat.widget.AppCompatImageButton;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.loopj.android.http.AsyncHttpResponseHandler;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import es.wolfi.passman.API.Credential;
+import es.wolfi.passman.API.CustomField;
+import es.wolfi.passman.API.File;
+
+/**
+ * {@link RecyclerView.Adapter} that can display a {@link File}.
+ * TODO: Replace the implementation with code for your data type.
+ */
+public class CustomFieldEditAdapter extends RecyclerView.Adapter<CustomFieldEditAdapter.ViewHolder> {
+
+    private final Credential credential;
+    private final List<CustomField> mValues;
+
+    public CustomFieldEditAdapter(Credential cred) {
+        credential = cred;
+        mValues = credential.getCustomFieldsList();
+    }
+
+    public String getCustomFieldsString() {
+        JSONArray customFields = new JSONArray();
+        for (CustomField cf : mValues) {
+            try {
+                customFields.put(cf.getAsJSONObject());
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return customFields.toString().replaceAll("\\\\/", "/");
+    }
+
+    public boolean addCustomField(CustomField customField) {
+        return mValues.add(customField);
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.fragment_credential_custom_field_item, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @SuppressLint("DefaultLocale")
+    public static String humanReadableByteCount(long bytes, boolean si) {
```

Comment:
There is already an implementation of this function [here](https://github.com/nextcloud/passman-android/blob/master/app/src/main/java/es/wolfi/utils/FileUtils.java#L374)

Reply:
fixed in b0f83244b829a59aed74bb30a44134a98c0509d2